src/middlewared/middlewared/plugins/device_/device_info_linux.py

```diff
@@ -133,28 +156,33 @@ def get_disk_details(self, block_device, disk, lshw_disks):
             'subsystem': os.path.realpath(os.path.join(disk_sys_path, 'device/subsystem')).split('/')[-1],
         })
 
-        disk['sectorsize'] = self.logical_sector_size(block_device.sys_name)
-
-        type_path = os.path.join(disk_sys_path, 'queue/rotational')
-        if os.path.exists(type_path):
-            with open(type_path, 'r') as f:
-                disk['type'] = 'SSD' if f.read().strip() == '0' else 'HDD'
-        else:
-            self.middleware.logger.error(
-                'Unable to retrieve %r disk rotational details at %s', disk['name'], type_path
-            )
+        if device_path in disks_data:
+            disk_data = disks_data[device_path]
 
-        if device_path in lshw_disks:
-            disk_data = lshw_disks[device_path]
+            # get type of disk and rotational rate (if HDD)
+            disk['type'] = 'SSD' if not disk_data['rota'] else 'HDD'
             if disk['type'] == 'HDD':
-                disk['rotationrate'] = disk_data['rotationrate']
+                disk['rotationrate'] = self.get_rotational_rate(device_path)
 
-            disk['ident'] = disk['serial'] = disk_data.get('serial', '')
-            disk['size'] = disk['mediasize'] = int(disk_data['size']) if 'size' in disk_data else None
-            disk['descr'] = disk['model'] = disk_data.get('product')
+            # get model and serial
+            disk['ident'] = disk['serial'] = disk_data.get('serial', '').strip()
+            disk['descr'] = disk['model'] = disk_data.get('model').strip()
+
+            # get all relevant size attributes of disk
+            disk['sectorsize'] = disk_data['log-sec']
+            disk['size'] = disk['mediasize'] = disk_data['size']
             if disk['size'] and disk['sectorsize']:
                 disk['blocks'] = int(disk['size'] / disk['sectorsize'])
 
+            # get lunid
+            if disk['tran'] == 'nvme':
+                disk['lunid'] = disk_data['wwn'].lstrip('eui').lower()
```

Comment:
We have the prefix `eui.` which we should be removing. Looking at `lsblk` output again, i don't think we need lower, `sg_vpd` identification flag returns them capitalised, lsblk seems to be doing the right thing following what we have been doing before.